content/_code-samples/send-xrp/SendXrp.java

```diff
@@ -0,0 +1,102 @@
+
+// Request current fee information from rippled
+FeeResult feeResult = xrplClient.fee();
+XrpCurrencyAmount openLedgerFee = feeResult.drops().openLedgerFee();
+
+// Get the latest validated ledger index
+LedgerIndex validatedLedger = xrplClient.ledger(LedgerRequestParams.builder().ledgerIndex(LedgerIndex.VALIDATED).build())
+  .ledgerIndex()
+  .orElseThrow(() -> new RuntimeException("LedgerIndex not available."));
+
+// Workaround for https://github.com/XRPLF/xrpl4j/issues/84
+UnsignedInteger lastLedgerSequence = UnsignedInteger.valueOf(
+  validatedLedger.plus(UnsignedLong.valueOf(4)).unsignedLongValue().intValue()
+  );
+
+// Construct a Payment
+Payment payment = Payment.builder()
+  .account(classicAddress)
+  .amount(XrpCurrencyAmount.ofXrp(BigDecimal.ONE))
+  .destination(Address.of("rPT1Sjq2YGrBMTttX4GZHjKu9dyfzbpAYe"))
+  .sequence(sequence)
+  .fee(openLedgerFee)
+  .signingPublicKey(testWallet.publicKey())
+  .lastLedgerSequence(lastLedgerSequence)
+  .build();
+System.out.println("Constructed Payment: " + payment);
+
+// Sign transaction -------------------------------------------------------------
+// Construct a SignatureService to sign the Payment
+PrivateKey privateKey = PrivateKey.fromBase16EncodedPrivateKey(testWallet.privateKey().get());
+SignatureService signatureService = new SingleKeySignatureService(privateKey);
+
+// Sign the Payment
+SignedTransaction<Payment> signedPayment = signatureService.sign(KeyMetadata.EMPTY, payment);
+System.out.println("Signed Payment: " + signedPayment.signedTransaction());
+
+// Submit transaction -----------------------------------------------------------
+SubmitResult<Transaction> submitResult = xrplClient.submit(signedPayment);
+System.out.println(submitResult);
+
+// Wait for validation ----------------------------------------------------------
+boolean transactionValidated = false;
+boolean transactionExpired = false;
+while (!transactionValidated && !transactionExpired) {
+  Thread.sleep(4 * 1000);
+  LedgerIndex latestValidatedLedgerIndex = xrplClient.ledger(
+      LedgerRequestParams.builder().ledgerIndex(LedgerIndex.VALIDATED).build()
+    )
+    .ledgerIndex()
+    .orElseThrow(() -> new RuntimeException("Ledger response did not contain a LedgerIndex."));
+
+  TransactionResult<Payment> transactionResult = xrplClient.transaction(
+    TransactionRequestParams.of(signedPayment.hash()),
+    Payment.class
+  );
+
+  if (transactionResult.validated()) {
+    System.out.println("Payment was validated with result code " + transactionResult.metadata().get().transactionResult());
+    transactionValidated = true;
+  } else {
+    boolean lastLedgerSequenceHasPassed = FluentCompareTo.
+      is(latestValidatedLedgerIndex.unsignedLongValue())
+      .greaterThan(UnsignedLong.valueOf(lastLedgerSequence.intValue()));
+    if (lastLedgerSequenceHasPassed) {
+      System.out.println("Payment was never validated and has expired.");
```

Comment:
This statement is too strong. You don't know for sure that the transaction has expired unless you also know that the server you submitted to has all the ledger history necessary, and this code does not check that. It's possible that the server you submitted to had a temporary desync and is missing the ledger in which the transaction _was_ validated.

Doing that check is probably out of the scope of this sample code, but the statement should not overpromise. Compare with the JS and Python print statements:

> If the transaction hasn't succeeded by now, it's expired

and

> max_ledger has passed. Last tx response: